dask/array/core.py

```diff
@@ -2657,14 +2679,19 @@ def store_chunk(x, out, index, lock, region, return_stored):
         lock.acquire()
     try:
         out[subindex] = np.asanyarray(x)
-    finally:
+    except:  # noqa: E722
         if lock:
             lock.release()
+        raise
```

Comment:
Need to make sure we always release the lock if there was an exception as we won't make it to `load_store_chunk` to release it.